server/webhook.go

```diff
@@ -169,33 +169,87 @@ func newWebhook(jwh *JiraWebhook, eventType string, format string, args ...inter
 	}
 }
 
-func (p *Plugin) GetWebhookURL(jiraURL string, teamID, channelID string) (subURL, legacyURL string, err error) {
-	cf := p.getConfig()
+func (wh *webhook) getConnection(p *Plugin, instance Instance, notification webhookUserNotification) (con *Connection, err error) {
+	var mattermostUserID types.ID
 
-	instanceID, err := p.ResolveWebhookInstanceURL(jiraURL)
+	// prefer accountId to username when looking up UserIds
+
+	if notification.jiraAccountID != "" {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraAccountID)
+	} else {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), notification.jiraUsername)
+	}
 	if err != nil {
-		return "", "", err
+		return
 	}
 
-	team, appErr := p.API.GetTeam(teamID)
-	if appErr != nil {
-		return "", "", appErr
+	// Check if the user has permissions.
+	con, err = p.userStore.LoadConnection(instance.GetID(), mattermostUserID)
+	if err != nil {
+		// Not connected to Jira, so can't check permissions
+		return
 	}
 
-	channel, appErr := p.API.GetChannel(channelID)
-	if appErr != nil {
-		return "", "", appErr
+	return
+}
+
+func (wh *webhook) CheckIssueWatchers(p *Plugin, instanceID types.ID) {
+	if len(wh.notifications) == 0 {
+		return
+	}
+
+	instance, err := p.instanceStore.LoadInstance(instanceID)
+	if err != nil {
+		// This isn't an internal server error. There's just no instance installed.
```

Comment:
> This isn't an internal server error. There's just no instance installed.

How do you know this is true? It looks like there are several kinds of errors that can happen in `LoadInstance`. Can we make it so the `LoadInstance` function returns a `nil` error, and `nil` for `instance`, so we know whether it was a real error if there is no instance?

Reply:
@vicky-demansol This request hasn't been addressed yet the comment was marked as resolved

Reply:
@mickmister we can use `error is kvstore error not found` but i dont know how to tell the loggers from webhook

Reply:
@mickmister @vicky-demansol is doing just that. But if it returns nil, do we log it here https://github.com/mattermost/mattermost-plugin-jira/blob/2325e8e9a354911bb7ed5a749f9f921429633633/server/webhook.go#L204